go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -503,10 +503,11 @@ func delServiceRules(service *kapi.Service, npw *nodePortWatcher) {
 	delGatewayIptRules(service, false)
 }
 
-func serviceUpdateNeeded(old, new *kapi.Service) bool {
+func serviceUpdateNotNeeded(old, new *kapi.Service) bool {
 	return reflect.DeepEqual(new.Spec.Ports, old.Spec.Ports) &&
 		reflect.DeepEqual(new.Spec.ExternalIPs, old.Spec.ExternalIPs) &&
 		reflect.DeepEqual(new.Spec.ClusterIP, old.Spec.ClusterIP) &&
```

Comment:
shouldn't ClusterIPs will always be populated even with single IP, do we still need ClusterIP ?